### PR TITLE
Contact deleting over API v2

### PR DESCRIPTION
**Pull request by rowanseymour**
Also updates the resthook subscribers endpoint to use new delete functionality

### CODE REVIEW

File: temba/api/v2/views.py [hunk 1]
```diff
@@ -182,11 +185,18 @@ class BaseAPIView(generics.GenericAPIView):
     Base class of all our API endpoints
     """
     permission_classes = (SSLPermission, APIPermission)
+    throttle_scope = 'v2'
```
Comment (rowanseymour) [inline temba/api/v2/views.py]: Moving these things into `BaseAPIView` from `ListAPIMixin` as they apply to POSTs and DELETEs as well as GETs

File: temba/api/v2/views.py [hunk 2]
```diff
@@ -298,13 +301,31 @@ def render_write_response(self, write_output, context):
...
+class DeleteAPIMixin(mixins.DestroyModelMixin):
     """
     Mixin for any endpoint that can delete objects with a DELETE request
     """
+    lookup_params = {'uuid': 'uuid'}
+
     def delete(self, request, *args, **kwargs):
         return self.destroy(request, *args, **kwargs)
 
+    def get_object(self):
```
Comment (rowanseymour) [inline temba/api/v2/views.py]: Regular DRF `get_object` expects object identifier in URL conf so overriding this to provide an easy way to look up objects by querystring params like "uuid"

File: temba/api/v2/views.py [hunk 3]
```diff
@@ -1826,10 +1878,14 @@ class ResthookSubscriberEndpoint(ListAPIMixin, CreateAPIMixin, DeleteAPIMixin, B
     write_serializer_class = ResthookSubscriberWriteSerializer
     pagination_class = CreatedOnCursorPagination
     throttle_scope = 'v2.api'
+    lookup_params = {'id': 'id'}
 
     def get_queryset(self):
         org = self.request.user.get_org()
-        return ResthookSubscriber.objects.filter(resthook__org=org, is_active=True).order_by('-created_on')
+        return self.model.objects.filter(resthook__org=org, is_active=True)
```
Comment (rowanseymour) [inline temba/api/v2/views.py]: Paginator does the ordering

**Comment by norkans7** [comment]
what happen when deleting a contact by urn for a contact that has many urns?

**Comment by rowanseymour** [comment]
@norkans7 it calls .release() which will detach all urns

**Comment by norkans7** [comment]
may be that should just remove the urn only and leave the contact with other urns

**Comment by rowanseymour** [comment]
The thinking here is that URN should be another way of identifying a contact - and work the same as UUID. If users want to remove a URN from a contact they can do that via a POST

**Comment by nicpottier** [comment]
Ya I think the behavior Rowan is describing is correct. The `DELETE` verb on the actions endpoint should delete the entire contact, not just the URN from a contact

**Comment by nicpottier** [comment]
yep

**Pull request merged**